tests/test_core/test_models.py

```
@@ -0,0 +1,260 @@
+    assert progress.end_time is None
+
+def test_config_model_full(sample_config_dict):
+    """Test full Config model validation."""
+    config = Config.model_validate(sample_config_dict)
+    assert config.years == [2023]
+    assert config.quarters == [1, 2]
+    assert config.proxy.enabled is False
+    assert config.download.pdfs is True
+    assert config.header_generator.use_generator is True
+
+@pytest.mark.parametrize("field,value,should_raise", [
```

Ruff (0.8.2)

Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)